frontend/public/components/utils/request-size-input.tsx

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable no-undef, no-unused-vars */
+import * as React from 'react';
+import { Dropdown } from '.';
+
+export class RequestSizeInput extends React.Component<RequestSizeInputProps> {
+  onChange = event => {
+    // Need this logic to handle the different input from dropdown vs input
+    let obj = {};
+    if (typeof event !== 'object') {
+      obj = { target: { name: `${this.props.name}Unit`, value: event } };
+    } else {
+      obj = event;
+    }
+    this.props.onChange(obj);
+  };
+
+  render() {
+    const ariaName = `${this.props.name}-help`;
+    const inputName = `${this.props.name}Value`;
+    const dropdownName = `${this.props.name}Unit`;
+    return (
+      <div className="form-group">
+        <div className="form-display-flex">
+          <input
+            className="form-control"
+            type="number"
+            onChange={this.onChange}
+            placeholder={this.props.placeholder}
+            aria-describedby={ariaName}
+            id={inputName}
+            name={inputName}
+            required={this.props.required}
+          />
+          <Dropdown
+            title={this.props.defaultRequestSizeUnit}
+            selectedKey={this.props.defaultRequestSizeUnit}
+            id={dropdownName}
+            name={dropdownName}
+            className="btn-group"
+            items={this.props.dropdownUnits}
+            onChange={this.onChange}
+            required={this.props.required}
+          />
+        </div>
+      </div>
+    );
+  }
+}
+
+export type RequestSizeInputProps = {
+  placeholder?: string;
```

Comment:
Should be separated by comma

Reply:
I think semicolons are appropriate here.